src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs

```diff
@@ -178,6 +178,19 @@ private async Task InvokeWorkerAsync(IProgress<CodeAnalysisProgress> progressTra
                         progressTracker,
                         cancellationToken).ConfigureAwait(false);
                 }
+
+                // Invoke suggested action executed callbacks in background, we don't want to block the UI thread on these callbacks.
+                _ = Task.Run(() =>
+                {
+                    // Ensure that the callbacks are executed serially by acquiring a lock on the SourceProvider.
```

Comment:
why do we care about this here?  why not have the callback itself serialize if necessary?

Reply:
We can do so as well - let me move this to the callback.

Reply:
Actually, there isn't a need to serialize in the callback either, so I have just removed this lock.